modules/nw-sriov-network-attachment.adoc

```diff
@@ -28,20 +34,41 @@ apiVersion: sriovnetwork.openshift.io/v1
 spec:
   networkNamespace: <target_namespace> <3>
   ipam: |- <4>
     ...
   vlan: <vlan> <5>
   resourceName: <sriov_resource_name> <6>
+  linkState: <link_state> <7>
+  maxTxRate: <max_tx_rate> <8>
+  minTxRate: <mix_rx_rate> <9>
+  vlanQoS: <vlan_qos> <10>
+  spoofChk: "<spoof_check>" <11>
+  trust: "<trust_vf>" <12>
+  capabilities: <capabilities> <13>
+  
 ----
 <1> Replace `<name>` with a name for the CR. The Operator will create a NetworkAttachmentDefinition CR with same name.
 <2> Specify the namespace where the SR-IOV Operator is installed.
-<3> Replace `<target_namespace>` with the namespace where the NetworkAttachmentDefinition CR will be created.
-<4> Specify a configuration object for the ipam CNI plug-in as a YAML block scalar. The plug-in manages IP address assignment for the attachment definition.
-<5> Replace `<vlan>` with a Virtual LAN (VLAN) ID for the additional network. The integer value must be from `0` to `4095`. The default value is `0`.
+<3> Optional. Replace `<target_namespace>` with the namespace where the NetworkAttachmentDefinition CR will be created. The default value is `openshift-sriov-network-operator`.
+<4> Optional. Specify a configuration object for the ipam CNI plug-in as a YAML block scalar. The plug-in manages IP address assignment for the attachment definition.
+<5> Optional. Replace `<vlan>` with a Virtual LAN (VLAN) ID for the additional network. The integer value must be from `0` to `4095`. The default value is `0`.
 <6> Replace `<sriov_resource_name>` with the value for the `.spec.resourceName` parameter from the SriovNetworkNodePolicy CR that defines the SR-IOV hardware for this additional network.
+<7> Optional. Replace `<link_state>` with the link state of Virtual Function (VF). Allowed value are `enable`, `disable` and `auto`.
+<8> Optional. Replace `<max_tx_rate>` with a maximum transmission rate, in Mbps, for the VF.
+<9> Optional. Replace `<min_tx_rate>` with a minimum transmission rate, in Mbps, for the VF. This value should always be less than or equal to Maximum transmission rate.
++
+NOTE: Intel NICs do not support the `minTxRate` parameter.
+<10> Optional. Replace `<vlan_qos>` with an IEEE 802.1p priority level for the VF. The default value is `0`.
+<11> Optional. Replace `<spoof_check>` with the spoof check mode of the VF. The allowed values are the strings `"on"` and `"off"`.
++
+IMPORTANT: You must enclose the value you specify in quotes or the CR will be rejected by the SR-IOV Network Operator.
+<12> Optional. Replace `<trust_vf>` with the trust mode of the VF. The allowed values are the strings `"on"` and `"off"`.
++
+IMPORTANT: You must enclose the value you specify in quotes or the CR will be rejected by the SR-IOV Network Operator.
+<13> Optional. Replace `<capabilities>` with the capabilities to configure for this network. You can specify `"{ "ips": true }"` to enable IP address support or `"{ "mac": true }"` to enable MAC address support.
```

Comment:
@pliurh, is it correct to say `capabilities` enables IP address or MAC address support? Can it be both? And if this is enabled, what does that actually mean? Why would a user want to enable one of these capabilities? What is the behavior if neither is enabled?

Thanks!

Reply:
You can have both. With MAC or IP capabilities`enabled` means you can specify the IP and/or MAC address in the pod manifests. If neither is enabled, then CNI will give you an automatic generated IP and MAC address. It's a new feature of Multus. There shall be a separated document to descript this in detail. I think it is better to have a link to that doc. @dougbtv Could you share the PR for this feature?

Reply:
@pliurh, there's a tooling limitation that currently prevents us from having any `xref` in modules. So we can't actually link to that content directly.

If it's modularized, we could include it in the SR-IOV Network Operator assembly.

Reply:
> @pliurh, is it correct to say `capabilities` enables IP address or MAC address support? 

`capabilities` enables static assignment of IP and MAC address. 
For static IP support, static ipam should be used (e.g. use static ipam type in SR-IOV net-attach-def).
For static MAC support in SR-IOV case, SR-IOV CNI needs to have the ability to accept the given MAC address and set it to actual device.

Can it be both? 

Yes, it can have both MAC and IP capabilities defined in `capabilities` field.

And if this is enabled, what does that actually mean? 

If it's enabled, Multus will parse it and call libcni to replace the capabilites with MAC and IP addresses defined in Pod Annotation(by user), then pass those addresses to delegated CNI via CNI rumtimeConfig. SR-IOV CNI or static IPAM plugin should(already) support CNI rumtimeConfig.

Why would a user want to enable one of these capabilities? 

User wants to configure static IP or/and MAC addresses.

What is the behavior if neither is enabled?

If neither is enabled, then static MAC or/and IP configured in Pod annotation will not be applied to pod interface.
(This is not 100% true, as we support legacy CNI args configuration in SRIOV and IPAM plugin, which will appy static MAC and IP to interface if they are passed down as CNI args, however, this mode will be deprecated in the future releases.)

> 
> Thanks!